### PR TITLE
RFC on UX: Warn if user is paying before about [x] months when it is due

### PR DESCRIPTION
**Pull request by weedySeaDragon**
## PT Story:  Warn if user is paying, but are already paid up > 2 months out
https://www.pivotaltracker.com/story/show/166273184

## Changes proposed in this pull request:
1.  Implemented the number of days to use to determine when a user "should" pay in the Application Configuration. You can log in as an admin and see it (and change it) there.

2.  I changed the layout of the member's membership payment term and the company h-markt branding license term.  See the screenshots below for details.

3.  If they do not need to make a payment, a modal comes up when they click the payment button.  The modal explains that they don't need to make a payment, but they can.

Please give me feedback on the layout, design, and interaction.

## Screenshots:

I've done screenshots with and without annotations.   For some of these I've first shown the screenshot _without_ annotations so that it's easier to see the layout and design.  Below it is a screenshot with annotations.  It does make this long with all the images.

### _Member -  Membership: No payment is due now:_

member-no-payment-needed

anntd-ae-member-no-payment-needed

---

_Tooltip for the payment button:_
---

anntd-ae-member-no-payment-needed-tooltip

---

_Modal that comes up when payment button is clicked: (In the 2nd screenshot, the underlined in red shows the words that are different from the Company branding license modal below.)_
---

modal-no-pay-needed-member

anntd-ae-modal-no-pay-needed-member

---
---

_When logged in as an admin:_
---

member-admin-no-payment-needed

anntd-ae-member-admin-no-payment-needed

---
---
---

### _Member -  Membership: Payment should be made now:_

member-payment-needed-tooltip

anntd-ae-member-payment-needed-tooltip

---
---

### _Company -  H-Markt Branding license: No payment is due now:_

company-no-payment-needed

anntd-ae-company-no-payment-needed

---
---

_Tooltip for the payment button:_
---

anntd-ae-company-no-payment-needed-tooltip

---

_Modal that comes up when payment button is clicked: (In the 2nd screenshot, the underlined in red shows the words that are different from the Member membership modal above.)_
---

modal-no-pay-needed-company

anntd-ae-modal-no-pay-needed-company

---
---

_When logged in as an admin:_
---

anntd-ae-company-admin-no-payment-needed

---
---
---

### _Company -  H-Markt Branding license: Payment should be made now:_

company-payment-needed-tooltip

anntd-ae-company-payment-needed-tooltip

---

## Ready for review:
@AgileVentures/shf-project-team

### CODE REVIEW

File: config/locales/sv.yml [hunk 1]
```diff
@@ -322,14 +325,20 @@ sv:
     kommun: *kommun
     none: *none_n
 
+
+  payors:
+    status: Status
+    no_payment_due_now: &no_payment_due_now No payment is due now.
```
Comment (thesuss) [inline config/locales/sv.yml]: Ingen betalning behöver göras nu.

File: config/locales/sv.yml [hunk 2]
```diff
@@ -322,14 +325,20 @@ sv:
...
+    term_paid_through: &term_paid_through Paid through
```
Comment (thesuss) [inline config/locales/sv.yml]: Betald till och med

**Review by thesuss — CHANGES_REQUESTED** [review]

**Review by patmbolger — COMMENTED** [review]
Looks great.  I didn't examine the code since it appears that one or more other PR changes are included in this (?)

**Comment by weedySeaDragon** [comment]
UX approved.  I'll do the implementation in a separate PR (or PRs).   (Which is what I originally said I'd do.  I just got sooooo excited about the implementation that I started adding it to this PR. ;-) )

**Pull request closed without merging**